### PR TITLE
Adds nighttime

### PR DESCRIPTION
**Pull request by Bamhalazam**
## About The Pull Request

The time has come. Amethyn's planetary rotation has restarted. The planet is no longer permanently locked with one side to the sun!

If the shift starts earlier than 08:00 in the morning and 20:00 at night, the outside will be, as expected, dark and full of surprises.

Currently this PR is in beta and all feedback on it would be greatly appreciated. Vision lines in the forest during night are vastly changed. The trees and rocks in the deep forest have also been moved from their original places to avoid visual bugs.
</summary>
<hr>

## Changelog
:cl:
add: Added Day/Night Cycle
tweak: tweaked vision code
imageadd: added nighttime sprites for trees
imagedel: deleted some icons and images
code: changed some time code
server: Added the Day/Night cycle which **NEEDS TO BE TESTED FOR PERFORMANCE**
:cl:

### CODE REVIEW

**Comment by cdb-is-not-good** [comment]
shitposting aside cool PR, I assume this is tested on your box with no serious issue?

**Comment by Bamhalazam** [comment]
> shitposting aside cool PR, I assume this is tested on your box with no serious issue?

Most of the major issues have been taken care of, it has some rough edges which this will try and even out :P 
I'll be merging a few commits over the course of the next day and will let you know when it's ready for a test.

**Comment by cdb-is-not-good** [comment]
👍

**Comment by benj8560** [comment]
sick as fuck but I see one issue personally. I'm not included in this predicament but I assume many others are. We just changed the server to properly convey time moving between rounds based on IRL time. Meaning if you only ever get time to play soj around a particular time you'll never get to see these changes all to much.

**Comment by Bamhalazam** [comment]
> sick as fuck but I see one issue personally. I'm not included in this predicament but I assume many others are. We just changed the server to properly convey time moving between rounds based on IRL time. Meaning if you only ever get time to play soj around a particular time you'll never get to see these changes all to much.

I've talked to Spock about this and expressed my concern. I think it should be brought up again on a wider forum, personally I don't support that.

**Comment by Bamhalazam** [comment]
@cdb-is-not-good @Trilbyspaceclone 

It is ready.

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -275,24 +275,33 @@
...
+	if (oldarea.has_gravity != newarea.has_gravity)
+		if (newarea.has_gravity == 1 && !MOVING_DELIBERATELY(L))
 			thunk(L)
 		L.update_floating()
 
+	if (newarea.is_forest)
+		L.see_in_dark = 100
+	else
+		L.update_sight()
+
```
Comment (Trilbyspaceclone) [inline code/game/area/areas.dm:300]:
```suggestion
	if(ishuman(L))
		var/mob/living/carbon/human/H = L handle_vision
		if (newarea.is_forest)
			H.see_in_dark = 100
		else
			H.species.handle_vision(H)
			H.update_equipment_vision()

```

**Comment by cdb-is-not-good** [comment]
unmerged on request from author.

**Comment by cdb-is-not-good** [comment]
Marking as draft till this is ready to try again.

**Comment by Bamhalazam** [comment]
I'm going to take another week to include new map changes in this and then ping y'all again when it's ready.

**Pull request closed without merging**